### DISCUSSION TITLE
Progress ability missing on StreamClient.GetAsync

### DISCUSSION THREAD

**Discussion by gcadmes-extron** [Tyrrrz/YoutubeExplode#761] [Q&A] [answered]
I can see `StreamClient.DownloadAsync` provides an `IProgress<T>` argument, my question is, why wouldn't `StreamClient.GetAsync` provide the same progress ability?

Aside from Windows and macOS support...for anyone wanting to add support for iOS & Android (Which I'm nearly done), will not be able to use the[ CommunityToolkit.Maui.Storage.IFileSaver](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/essentials/file-saver?tabs=android) object. 
This is mostly due to the `IFileSaver.SaveAsync` expecting a `System.IO.Stream`.
My plan was to use the `ValueTask<Stream>` that is returned from `StreamClient.GetAsync`, which I am able to do, however, it would be nice for the user to see the progress from that specific API usage.
Thoughts?

**Comment by Tyrrrz** [comment]
Because `GetAsync(...)` completes pretty much immediately, the caller is responsible for copying bytes (and thus, reporting progress)

**Reply by gcadmes** [reply]
@Tyrrrz  thanks for the quick reply. Much appreciated! 
I suppose I will need to overload the `StreamClient.DownloadAsync `and instead of creating a destination `FileStream`, I will pass in a platform agnostic `Stream `instance.

**Reply by Tyrrrz** [reply, marked as answer]
What you're looking for is `StreamClient.CopyToAsync` I think
